Binary XML file line #18: You must supply a layout_width attribute.

Hi, I just copied from your instruction and then compiled. I've got some error message like this below 

Caused by: java.lang.UnsupportedOperationException: Binary XML file line #18: You must supply a layout_width attribute.
                                                                    at android.content.res.TypedArray.getLayoutDimension(TypedArray.java:709)
                                                                    at android.view.ViewGroup$LayoutParams.setBaseAttributes(ViewGroup.java:6889)
                                                                    at android.view.ViewGroup$MarginLayoutParams.<init>(ViewGroup.java:7071)
                                                                    at android.widget.FrameLayout$LayoutParams.<init>(FrameLayout.java:446)
                                                                    at android.widget.FrameLayout.generateLayoutParams(FrameLayout.java:386)
                                                                    at android.widget.FrameLayout.generateLayoutParams(FrameLayout.java:385)
                                                                    at android.view.LayoutInflater.inflate(LayoutInflater.java:502)
                                                                    at android.view.LayoutInflater.inflate(LayoutInflater.java:423) 
                                                                    at com.karumi.expandableselector.ExpandableSelector.initializeButton(ExpandableSelector.java:277) 
                                                                    at com.karumi.expandableselector.ExpandableSelector.renderExpandableItems(ExpandableSelector.java:231) 
                                                                    at com.karumi.expandableselector.ExpandableSelector.showExpandableItems(ExpandableSelector.java:92) 
                                                                    at indie.kcalulator.ListsActivity.initializeSizesExpandableSelector(ListsActivity.java:84) 
                                                                    at indie.kcalulator.ListsActivity.onCreate(ListsActivity.java:69) 
                                                                    at android.app.Activity.performCreate(Activity.java:6237) 
                                                                    at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1107) 
                                                                    at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2369) 
                                                                    at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2476) 
                                                                    at android.app.ActivityThread.-wrap11(ActivityThread.java) 
                                                                    at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1344) 
                                                                    at android.os.Handler.dispatchMessage(Handler.java:102) 
                                                                    at android.os.Looper.loop(Looper.java:148) 
                                                                    at android.app.ActivityThread.main(ActivityThread.java:5417) 
                                                                    at java.lang.reflect.Method.invoke(Native Method) 
                                                                    at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:726) 
                                                                    at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:616) 

Could you tell me how do I fix this because this error message told me that your code is not declare a layout_width attribute on android view. 

I suppose that tools:ignore="RequiredSize" in expandable_item_button layout doesn't work 

Thanks in advance.

Hi, I just copied from your instruction and then compiled. I've got some error message like this below

java.lang.RuntimeException: Binary XML file line #18: You must supply a layout_width attribute.
                                                                                   at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2205)
                                                                                   at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2255)
                                                                                   at android.app.ActivityThread.access$800(ActivityThread.java:142)
                                                                                   at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1203)
                                                                                   at android.os.Handler.dispatchMessage(Handler.java:102)
                                                                                   at android.os.Looper.loop(Looper.java:136)
                                                                                   at android.app.ActivityThread.main(ActivityThread.java:5118)
                                                                                   at java.lang.reflect.Method.invokeNative(Native Method)
                                                                                   at java.lang.reflect.Method.invoke(Method.java:515)
                                                                                   at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:790)
                                                                                   at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:606)
                                                                                   at dalvik.system.NativeStart.main(Native Method)
                                                                                Caused by: java.lang.RuntimeException: Binary XML file line #18: You must supply a layout_width attribute.
                                                                                   at android.content.res.TypedArray.getLayoutDimension(TypedArray.java:492)
                                                                                   at android.view.ViewGroup$LayoutParams.setBaseAttributes(ViewGroup.java:5963)
                                                                                   at android.view.ViewGroup$MarginLayoutParams.<init>(ViewGroup.java:6132)
                                                                                   at android.widget.FrameLayout$LayoutParams.<init>(FrameLayout.java:615)
                                                                                   at android.widget.FrameLayout.generateLayoutParams(FrameLayout.java:559)
                                                                                   at android.widget.FrameLayout.generateLayoutParams(FrameLayout.java:56)
                                                                                   at android.view.LayoutInflater.inflate(LayoutInflater.java:480)
                                                                                   at android.view.LayoutInflater.inflate(LayoutInflater.java:397)
                                                                                   at com.karumi.expandableselector.ExpandableSelector.initializeButton(ExpandableSelector.java:279)
                                                                                   at com.karumi.expandableselector.ExpandableSelector.renderExpandableItems(ExpandableSelector.java:231)
                                                                                   at com.karumi.expandableselector.ExpandableSelector.showExpandableItems(ExpandableSelector.java:92)
                                                                                   at com.dxtre.www.detallitossabores.MainActivity.onCreate(MainActivity.java:36)
                                                                                   at android.app.Activity.performCreate(Activity.java:5275)
                                                                                   at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1087)
                                                                                   at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2169)
                                                                                   at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2255) 
                                                                                   at android.app.ActivityThread.access$800(ActivityThread.java:142) 
                                                                                   at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1203) 
                                                                                   at android.os.Handler.dispatchMessage(Handler.java:102) 
                                                                                   at android.os.Looper.loop(Looper.java:136) 
                                                                                   at android.app.ActivityThread.main(ActivityThread.java:5118) 
                                                                                   at java.lang.reflect.Method.invokeNative(Native Method) 
                                                                                   at java.lang.reflect.Method.invoke(Method.java:515) 
                                                                                   at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:790) 
                                                                                   at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:606) 
                                                                                   at dalvik.system.NativeStart.main(Native Method) 

--- a/README.md
+++ b/README.md
@@ -104,6 +104,8 @@ You can easily customize the appearance of your ``ExpandableItem`` instances. Ju
 
 ```
 
+***Remember that declaring a ``ExpandableItemStyle`` as described before with the layout height and width is mandatory to avoid inflation errors.***
+
 The resources you can show in the Button/ImageButton widgets automatically added to ``ExpandableSelector`` are:
 
 * Background resource identifier configured as Button/ImageButton background.

